src/pvt.c

```diff
@@ -278,6 +278,163 @@ static u8 filter_solution(gnss_solution* soln, dops_t* dops)
   return 0;
 }
 
+/* Checks pvt_iter residuals; can detect erroneous input measurements.
+ * Returns true if residual is sufficiently small.
+ */
+static bool residual_test(u8 n_used, double omp[n_used],
+                          const double rx_state[],
+                          double *residual)
+{
+  /* Very liberal threshold. Typical range 20 - 120 */
+#define PVT_RESIDUAL_THRESHOLD 3000
+
+  /* Need to add clock offset to observed-minus-predicted calculated by last
+   * iteration of pvt_solve before computing residual. */
+  for (int i = 0; i < n_used; i++) {
+    omp[i] -= rx_state[3];
+  }
+  double norm = vector_norm(n_used, omp);
+  if (residual) {
+    *residual = norm;
+  }
+  return norm < PVT_RESIDUAL_THRESHOLD;
+}
+
+/* Iterates pvt_solve until it converges or PVT_MAX_ITERATIONS is reached.
+ * Return value:
+ *   0: solution converged
+ *  -1: solution failed to converge
+ *
+ *  Results stored in rx_state, omp, H
+ */
+static s8 pvt_iter(double rx_state[],
+                   const u8 n_used,
+                   const navigation_measurement_t nav_meas[n_used],
+                   double omp[n_used],
+                   double H[4][4])
+{
+  /* Reset state to zero */
```

Comment:
Might want to clarify that this resets everything except the position part of the state.